fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/join/JoinLAsscomHelper.java

```diff
@@ -165,6 +166,7 @@ private Pair<List<NamedExpression>, List<NamedExpression>> getProjectExprs() {
     private LogicalJoin<GroupPlan, GroupPlan> newBottomJoin() {
         return new LogicalJoin(
                 bottomJoin.getJoinType(),
+                bottomJoin.getHashJoinPredicates(),
```

Comment:
bottomJoin.getHashJoinPredicates() must be empty, because this rule is ahead of FindHashConditionForJoin.
to make it more clear, I will use `ImmutableList.of()`

Reply:
yes, we should move `newBottomJoinOnCondition` to  `hashJoinPredicates`